### PR TITLE
Add/hub+spoke

### PR DESCRIPTION
**Pull request by jk-ans**
# Description

Please provide a summary of your changes and the reason the changes are being made.

Updating Bastion Module based on Tyla's work. Made some changes to the example during testing. Confirmed example is deploying correctly.

Please select the option that is relevant.

  - [X] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [X] This change requires a documentation update

# Checklist:

  - [X] My code follows the style guidelines of this project
  - [X] I have performed a self-review of my code
  - [X] I have commented my code, particularly in hard-to-understand areas
  - [X] I have made corresponding changes to the documentation
  - [X] My changes generate no new warnings e.g. code analysis tooling or general use of the changed code

### CODE REVIEW

**Review by tb-ans — APPROVED** [review]
Think we're good to introduce this into the repository. Suggest that we revisit this at a future date and consider adjustments to be made, for example adding extra diagnostic configurations, assessing RT/NSG allocations and additional config options or subnets with more predefined configurations like we've done with VNG.

**Comment by jk-ans** [comment]
> Think we're good to introduce this into the repository. Suggest that we revisit this at a future date and consider adjustments to be made, for example adding extra diagnostic configurations, assessing RT/NSG allocations and additional config options or subnets with more predefined configurations like we've done with VNG.

Aye, agreed!

File: azure/spoke/main.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  resource_group_name = azurerm_resource_group.main.name
+  tags                = var.tags
+}
+
+resource "azurerm_route" "main_default" {
+  count = var.route_table_name != null ? 1 : 0
+
+  name                = "default"
+  resource_group_name = azurerm_resource_group.main.name
+  route_table_name    = azurerm_route_table.main[count.index].name
```
Comment (djackson-ans) [inline azure/spoke/main.tf:51]: `count.index` is not needed here if the value can only ever be `0`.

File: azure/spoke/examples/basic/main.tf [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "snet-default" = {
+      prefix = "10.0.0.0/24"
+    }
+  }
+
+  network_security_group_name = "nsg-${local.resource_prefix}"
+  route_table_name            = "rt-${local.resource_prefix}"
+  default_route_ip            = "192.168.0.1"
+
+  # hub_virtual_network_id = "/subscriptions/CHANGEME/resourceGroups/CHANGEME/providers/Microsoft.VirtualNetwork/CHANGEME"
```
Comment (djackson-ans) [inline azure/spoke/examples/basic/main.tf:43]: is this still needed?

**Comment by djackson-ans** [comment]
@jk-ans - Please review when you can

**Pull request merged**